### PR TITLE
Upgrade/emberjs 1.13.2

### PR DESCRIPTION
**Pull request by peec**
This PR will upgrade the ember version to 1.13.2 and also include changes to make paper compatible with 1.13.2.

Specifically this is done:

- Change to 1.13.2 of emberjs.
- ~~The use of nearestOfType (`paper-sidenav-toggle` component) does not seem to work in 1.13.2. Instead a global event bus is used here. The event bus can come handy for other components as well. The event bus principle is explained quite nicely [here](http://www.thesoftwaresimpleton.com/blog/2015/04/27/event-bus/). As of such one can now trigger from any component:~~

    ```javascript
        paperEventBus: Ember.inject.service('paper-eventbus'),
        someAction: function () {
            this.get("paperEventBus").publish('paper:toggle-sidenav');
            // or 
            this.get("paperEventBus").publish('paper:expand-sidenav'); 
        }
    ```
    ~~Obviously this event-bus is global, so it should only be used for global components. One could also make these events non-global by adding attribute to the event name. i.e. `.publish('paper:toggle-sidenav-' + this.get("menuhandle")')`. Though I do not think this is necessary for the sidenav component.~~
    **EDIT**: using parentView now.

- Updated `{{#if template}}` to respective `{{#if hasBlock}}`. ( With respective mixin to add support for pre 1.13.X releases).
- Updated `ember-prism` to 0.0.5.

### CODE REVIEW

**Comment by miguelcobain** [comment]
@peec Please read https://github.com/emberjs/ember.js/issues/11170

tl;dr
- Right now, we should use `parentView`
- In the future we may be doing stuff like this:
```hbs
{{#paper-nav-container as |c|}}
  {{#c.paper-sidenav}} {{!-- sidenav here --}} {{/c.paper-sidenav}}
  {{#c.paper-content}} {{!-- content here --}} {{/c.paper-content}}
{{/paper-nav-container}}
```

Can you please refactor this PR to use `parentView` instead?

**Comment by miguelcobain** [comment]
Also, does using `hasBlock` break when using Ember 1.12?
Check #86.

**Comment by peec** [comment]
@miguelcobain 

I have now done the following:
- **hasBlock:** Created backwards compatible mixin so it doesn't break pre - 1.13.X releases.
- Removed service principle from commits and added support for **parentView**

**Comment by miguelcobain** [comment]
@peec I think `nearestOfType` works fine now.

**Comment by peec** [comment]
@miguelcobain 

Good idea, I tested with nearestOfType and it works fine now. 

Added:
- nearestOfType support again
- Check if hasBlock exist, if it does use it otherwise use this.get("template")

**Comment by miguelcobain** [comment]
Please remove `app/services/paper-eventbus.js`.

**Comment by miguelcobain** [comment]
Thanks @peec!

I opted to keep ember 1.12 for now. That should affect people's projects, because ember-paper is now compatible with 1.13 as well.

When we move to ember-cli 1.13.1 we use its ember version.
